### PR TITLE
#312 - Update README for changes to component lifecycle.

### PR DESCRIPTION
**Pull request by treshugart**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -377,10 +374,10 @@ skate('my-element', {
 The component lifecycle consists several callbacks:
 
 - `created`
```
Comment (bqx) [inline README.md]: It might be good to also indicate that the prototype/properties are set up here

**Comment by bqx** [comment]
:+1: lgtm

**Comment by marcoms** [comment]
By the way the table of contents still references the old template function

**Comment by alexreardon** [comment]
Was a decision made about going down the function composition path?

**Comment by treshugart** [comment]
@alexreardon as discussed IRL, yeah. Staying declarative unless it's absolutely necessary. We're pretty confident that this is the final big lifecycle change and should facilitate any templating / component route people would want to take.

**Comment by treshugart** [comment]
Done, done, done and done :)

**Comment by treshugart** [comment]
@bqx @marcoms I've also updated this removing a lot of the wordy stuff around the lifecycle in hopes that the ordered list is good enough along with the following description and `attribute` description.

I've also removed the wordy templating stuff. Even if this gets added back in (see #321) then the specifics around integration should probably be left out of the readme and put in the website or in the form of a blog / tutorial or something.

**Pull request merged**